addon/waiter-manager.ts

```diff
@@ -53,6 +53,10 @@ export function getWaiters(): Waiter[] {
  * @private
  */
 export function _reset(): void {
+  for (let waiter of getWaiters()) {
```

Comment:
this fixes a bug where if there are multiple calls to `waitForPromise`, `getPendingWaiterState()` _always_ returns no waiters. Because, `PROMISE_WAITER` is defined in module-space, is is not garbage collected during `WAITERS.clear()` (below). Re-setting `isRegistered` to false here allows the `PROMISE_WAITER` to be re-registered in the `WAITERS` `Map` during the next test.

Reply:
Nice find. Thanks for this.